spacer/task_utils.py

```diff
@@ -61,102 +65,316 @@ def check_extract_inputs(image: Image,
                 f" valid range of 0-{im_width - 1}.")
 
 
+class SplitMode(Enum):
+    """
+    How to split annotations between train, ref, and val sets.
+    """
+    # Each feature vector's points all go into train, or all go into ref, or
+    # all go into val.
+    #
+    # The rationale behind this mode is to have greater separation between
+    # training data and evaluation data, whether it's during the calibration
+    # process (train vs. ref) or during evaluation of the final classifier
+    # (train vs. val).
+    # Here we assume the imagery is 'more different' when going across feature
+    # vectors, as opposed to staying within the same feature vector. When
+    # training and evaluation data are 'more different', the result is more
+    # useful.
+    # Thus, this mode can improve usefulness of calibration, and rigor of the
+    # evaluation results.
+    # However, the annotation count may not end up precisely balanced
+    # between train/ref/val as desired, particularly when the feature vector
+    # size is comparable to the set size. For example, if each feature vector
+    # has 100 points, and the target ref-set size is 450, then the best we can
+    # do is giving the ref set either 400 or 500 points.
+    VECTORS = 'vectors'
+    # The split is done on an individual point basis, so a single
```

Comment:
Does this mean we are duplicating featurevectors  ?

Reply:
No - for example, if you have a feature vector with 20 point-features, then you might have 16 going to train, 2 going to ref, and 2 going to val. I can add an example like that in the comment here.

Reply:
Added that example.